neuralpp/symbolic/sympy_expression.py

```diff
@@ -329,22 +335,28 @@ def native_arguments(self) -> Tuple[sympy.Basic, ...]:
 
     @staticmethod
     def from_sympy_function_and_general_arguments(sympy_function: sympy.Basic, arguments: List[Expression]) -> \
-            SymPyFunctionApplication:
+            SymPyExpression:
```

Comment:
Since the function can possibly evaluate the SymPy object, it does not necessarily return a function application now.